tests/lib/rules/no-throw-literal.js

```diff
@@ -23,37 +23,48 @@ eslintTester.addRuleTest("lib/rules/no-throw-literal", {
         "throw new Error();",
         "throw new Error('error');",
         "throw Error('error');",
-        "throw {};",
-        "throw [];",
         "var e = new Error(); throw e;",
-        "try {throw new Error();} catch (e) {throw e;};"
+        "try {throw new Error();} catch (e) {throw e;};",
+        "throw a;", // Identifier
+        "throw foo();", // CallExpression
+        "throw new foo();", // NewExpression
+        "throw foo.bar;", // MemberExpression
+        "throw foo[bar];", // MemberExpression
+        "throw foo = new Error();", // AssignmentExpression
+        "throw 1, 2, new Error();", // SequenceExpression
+        "throw 'literal' && new Error();", // LogicalExpression (right)
+        "throw new Error() || 'literal';", // LogicalExpression (left)
+        "throw foo ? new Error() : 'literal';", // ConditionalExpression (consequent)
+        "throw foo ? 'literal' : new Error();", // ConditionalExpression (alternate)
+        { code: "throw tag `${foo}`;", ecmaFeatures: {templateStrings: true} }, // TaggedTemplateExpression
+        { code: "function* foo() { var index = 0; throw yield index++; }", ecmaFeatures: {generators: true} } // YieldExpression
     ],
     invalid: [
         {
             code: "throw 'error';",
             errors: [{
-                message: "Do not throw a literal.",
+                message: "Did not throw an Error object.",
```

Comment:
I think this message is a bit too mysterious. It really doesn't matter if you throw an `Error` object or another object, we just don't want them throwing a literal or primitive. Something like, "Expected an object to be thrown" may be clearer.